pkg/monitoring/gauge.go

```diff
@@ -65,40 +48,23 @@ func newGauge(o options) *gauge {
 
 func (f *gauge) Record(value float64) {
 	f.runRecordHook(value)
-	// TODO: https://github.com/open-telemetry/opentelemetry-specification/issues/2318 use synchronous gauge so we don't need to deal with this
-	f.attributeSetsMutex.Lock()
-	// Special case: we lazy-load the non-labeled value. This ensures that metrics which should always have labels do not end up with a un-labeled zero-value
-	// If a metric really requires `metric{} 0`, they can explicitly call .Record(0).
-	if f.currentGaugeSet == nil {
-		f.currentGaugeSet = &gaugeValues{}
-		f.attributeSets[attribute.NewSet()] = f.currentGaugeSet
+	if f.precomputedRecordOption != nil {
```

Comment:
Can this check be deleted?

Reply:
No, this is only used after With() is called. You can have no options

Reply:
I mean when `f.precomputedRecordOption == nil`, `f.precomputedRecordOption...` is also ok.

Reply:
I see, I might haev done it to reduce allocations. let me check

Reply:
Nope... good call